### PR TITLE
feat: migrate from terraform cloud to github oidc for terraform deploy workflow

### PR DESCRIPTION
**Pull request by NithinKuruba**
Fixes [SSOTEAM-912](https://bcdevex.atlassian.net/browse/SSOTEAM-912)

### CODE REVIEW

File: docs/github-oidc-for-tf-deploy.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+- Select `token.actions.githubusercontent.com` identity provider
+- Enter `repo:bcgov/sso-requests:ref:refs/heads/<branch>` under audience
+- Enter `bcgov` under GitHub org and click next
+- Add required policies and click next
+- Click create role
+- Edit the role and select `Trust Relationships`
+- Update the trusted entities
+
+  ```json
+  <!-- Update `account-number`, `branch` before using-->
```
Comment (thegentlemanphysicist) [inline docs/github-oidc-for-tf-deploy.md:17]: where do we get the account number?

File: docs/github-oidc-for-tf-deploy.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+  ```yaml
+  - name: Configure AWS Credentials
+    uses: aws-actions/configure-aws-credentials@v2
+    with:
+      role-to-assume: <role-arn-copied-from-aws>
+      aws-region: ca-central-1
+  ```
+
+- Add below permission if does not exist already
```
Comment (thegentlemanphysicist) [inline docs/github-oidc-for-tf-deploy.md:83]: Where is this permission added? I'm assuming from context it's added to the github action?

Comment (NithinKuruba) [reply docs/github-oidc-for-tf-deploy.md:83]: Yes it's in the github action.

File: docs/github-oidc-for-tf-deploy.md (hunk 1 above)
Comment (NithinKuruba) [reply docs/github-oidc-for-tf-deploy.md:17]: You can see the account number from aws login screen after successful azure idir login

**Review by thegentlemanphysicist — APPROVED** [review]

**Pull request merged**